chore: update docs for funds recovery (#498)
Consensys/starknet-snap: docs/tutorial-resolving-stuck-funds.md

> # Tutorial: Resolving Stuck Funds Using the New StarkNet Snap Environment

Author: ```suggestion
# Tutorial: Resolving Stuck Funds Using the StarkNet Snap Recovery Environment
```

If your funds are stuck on a non-deployed Cairo 0 or Cairo 1 account, follow the steps below to resolve the issue.

---

> ## Preamble: Uninstall the Current StarkNet Snap

Reviewer: Most users may not understand the word "Preamble' and how it fits into the whole process.
Just to make it feel part of the process, can it be renamed to "**Step 1:**", and renumber the subsequent steps ?

Author: It was initially set up so that both Preamble and Final Step were shared between the two cases to avoid repetition. However, to ensure clarity and avoid confusion, I’ve updated both guides to be fully self-contained while still keeping the necessary steps consistent.

> 1. Open your wallet application.

Reviewer: We can reword to "Open your MetaMask wallet extension in the browser".

2. Navigate to the installed Snaps section.
3. Find the StarkNet Snap and select it.

> 4. Click on “Uninstall” or “Remove” to uninstall the StarkNet Snap.

Reviewer: Is it Uninstall or Remove ?
Lets put the actual Menu Label here.

Author: Remove indeed!

---

> ## Part 1: Resolving Funds Stuck on a Non-Deployed Cairo 0 Account

Reviewer: We should rename "Part 1" to "Scenario 1" (and likewise the other _Parts_ also)
I was under the impression that all the Parts were part of the steps to undergo. But in reality, its different scenarios that the User finds them in and needs to perform exclusively.
So "Scenario" would be a better term.

### Step 1: Use the Link to Force Cairo 0 Deployment
1. Open your browser and go to the following URL:  
   [Force Cairo 0 Deployment](https://staging.snaps.consensys.io/starknet?accountDiscovery=FORCE_CAIRO_0)
2. Connect your MetaMask wallet when prompted.

> 3. Follow the instructions to install a special version of the StarkNet Snap.

Reviewer: Can we mention the "special version" number here to make sure the User is doing it correctly ?

### Step 2: Fund the Cairo 0 Account and Deploy It
1. Fund your Cairo 0 account with a small amount of ETH.

> 2. Reload the page after funding the account.

Reviewer: Should the user press F5 to reload the browser ?

Author: Yes or the other option is to click on the url bar and press enter. I am wondering if F5 shortcut is a cross-browser standard ?

Author: I am adding (e.g. using the F5 key) as a hint

3. A pop-up will appear prompting you to **deploy your Cairo 0 account**.
   - Follow the on-screen instructions to deploy the account.
   - Confirm and authorize the transaction in your wallet.

### Step 3: Upgrade the Cairo 0 Account
1. Once the Cairo 0 account is deployed, another pop-up will appear prompting you to **upgrade the account**.
   - Follow the on-screen instructions to upgrade your account.
   - Confirm and authorize the transaction in your wallet.
2. After the upgrade, your Cairo 0 account will be fully functional, and you will be able to send funds.

### Step 4: Transfer Funds to Your Cairo 1 Account
1. Once your Cairo 0 account is functional, check your balance to confirm the funds are available.

> 2. Transfer all funds from the Cairo 0 account to your Cairo 1 account.  

Reviewer: Going back to the Preamble/Step1, will the Snap at the point in time be the Cairo1 address ?
Should we ask the user to note the address there before asking them to uninstall the Snap ? And then use that address here in this step ?

Author: User can get the Cairo 1 address following this link:  [Force Cairo 1 Deployment](https://staging.snaps.consensys.io/starknet?accountDiscovery=FORCE_CAIRO_1).

I am adding a mention like this, wdyt ? 

> 2. Transfer all funds from the Cairo 0 account to your Cairo 1 account.  
>    - Ensure you use the correct Cairo 1 address to avoid any loss of funds. In case you don't know your Cairo 1 account address you can get it from here [Force Cairo 1 Deployment](https://staging.snaps.consensys.io/starknet?accountDiscovery=FORCE_CAIRO_1).
> **In case you don't know your Cairo 1 account address you can get it from here [Force Cairo 1 Deployment](https://staging.snaps.consensys.io/starknet?accountDiscovery=FORCE_CAIRO_1).**

   - Ensure you use the correct Cairo 1 address to avoid any loss of funds.
3. Confirm and authorize the transaction in your wallet.

---

> ## Part 2: Resolving Funds Stuck on a Non-Deployed Cairo 1 Account

Reviewer 2: is this step verified?

as we plan to ask user to discover the Cairo 1 address first before he transfer from Cairo 0
then he transfer all from Cairo 0
then he deploy Cairo 1 

just make sure below step has same result as above

Author: Yes double checked it should be correct.

### Step 1: Use the Link to Force Cairo 1 Deployment
1. Open your browser and go to the following URL:  
   [Force Cairo 1 Deployment](https://staging.snaps.consensys.io/starknet?accountDiscovery=FORCE_CAIRO_1)
2. Connect your MetaMask wallet when prompted.
3. Follow the instructions to install a special version of the StarkNet Snap.

### Step 2: Deploy the Cairo 1 Account by Initiating a Transaction
1. Once connected, fund your Cairo 1 account with a small amount of ETH (if not already funded).
2. Initiate any transaction (e.g., sending tokens) from the Cairo 1 account.
3. This transaction will automatically deploy the Cairo 1 account.
4. Confirm and authorize the transaction in your wallet.

### Step 3: Transfer Funds from Cairo 0 to Cairo 1

> 1. Once your Cairo 1 account is deployed, return to the link for [Force Cairo 0 Deployment](https://staging.snaps.consensys.io/starknet?accountDiscovery=FORCE_CAIRO_0) to view your Cairo 0 account.

Author: ```suggestion
1. Once your Cairo 1 account is deployed, follow the link to [Force Cairo 0 Deployment](https://staging.snaps.consensys.io/starknet?accountDiscovery=FORCE_CAIRO_0) to view your Cairo 0 account.
```

2. Transfer all funds from the Cairo 0 account to the deployed Cairo 1 account.  
   - Ensure you enter the correct Cairo 1 address to avoid any loss of funds.
3. Confirm and authorize the transaction in your wallet.

---

> ## Final Step: Install the Production StarkNet Snap

Reviewer: Is this **mandatory** Final step to be done - irrespective whichever Part/Scenario the user finds themselves in ?

Author: Yes I moved this final step in both scenarios to make each scenario fully self-contained

1. Uninstall the special StarkNet Snap version by:
   - Opening your wallet application.
   - Navigating to the installed Snaps section.

>    - Finding the special StarkNet Snap and selecting "Uninstall" or "Remove."

Author: ```suggestion
   - Finding the StarkNet Snap and selecting "Remove" from the menu
```
